### PR TITLE
Ipeakfunction intensityerror

### PR DESCRIPTION
**Pull request by jmborr**
**Description of issue:**  
See [PD234](https://code.ornl.gov/sns-hfir-scse/diffraction/powder/powder-diffraction/-/issues/234#description) for details

Companion to PR #31585

**Description of work.**
- [x] `PeakFunctionIntegrator::integrateError` and unit test
- [x] `IPeakFunction::intensityError` and unit test
- [x] updated release notes
- [x] workbench test

**To test:**  
Run the following script in Mantid's workbench to compare the analytical and numerical values of the integrated intensity for a custom peak function.  
```python
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
from mantid.api import IPeakFunction, FunctionFactory
import matplotlib.pyplot as plt
import numpy as np
from mantid.kernel import logger

class MyPeak(IPeakFunction):
    r""" f(x) = A / [(x-c)^4 + a^4] """
    def init(self):
        self.declareParameter("Amplitude")
        self.declareParameter("Fwhm")
        self.declareParameter("Center")

    def functionLocal(self, xvals):
        center = self.getParameterValue("Center")
        fwhm = self.getParameterValue("Fwhm")
        amplitude = self.getParameterValue("Amplitude")
        return amplitude / ((xvals - center)**4 + fwhm**4)

    # required function overrrides
    def fwhm(self):
        return self.getParameterValue("Fwhm")

    def setFwhm(self, newFwhm):
        self.setParameter("Fwhm", newFwhm)
    
    def centre(self):
        return self.getParameterValue("Center")

    def setCentre(self, newCenter):
        self.setParameter("Center", newCenter)

    def height(self):
        return self.getParameterValue("Amplitude") / self.getParameterValue("Fwhm")**4

    def setHeight(self, newHeight):
        self.setParameter("Amplitude", newHeight * self.getParameterValue("Fwhm")**4)
        
    def intensity(self):
        r"""Integration of the peak function from minus-infinity to plus-infinity"""
        amplitude = self.getParameterValue("Amplitude")
        fwhm = self.getParameterValue("Fwhm")
        return np.pi * amplitude / (np.sqrt(2) * fwhm**3)

    def intensityErrorLocal(self):
        r"""Analytical error in the integration of the peak function from minus-infinity to plus-infinity
        due to uncertainties in the fit parameters"""
        amplitude = self.getParameterValue("Amplitude")
        amplitudeError = self.getError("Amplitude")
        fwhm = self.getParameterValue("Fwhm")
        fwhmError = self.getError("Fwhm")
        return self.intensity() * np.sqrt((amplitudeError/amplitude)**2 + (3*fwhmError/fwhm)**2)

FunctionFactory.subscribe(MyPeak)

f = MyPeak()
f.initialize()
# Compare the analytical and the numerical values as a function of the error size
for error_percent in (1, 5, 10):
    for i, (name, value) in enumerate([("Amplitude", 10.0), ("Fwhm", 2.0), ("Center", 1.0)]):
        f.setParameter(name, value)
        f.setError(i, value * error_percent / 100.0)  # essential the order Amplitude, then Fwhm, then Center
    analytical = f.intensityErrorLocal()
    numerical = f.intensityError()
    logger.error(f'percentual error = {error_percent}, analytical error = {analytical}, numerical error = {numerical}')
```

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by KedoKudo — DISMISSED** [review]
The branch builds and runs on the analysis cluster without triggering any issue.
The provided testing script generated expected results (the analytical error and numerical ones are similar, but not exactly the same)
```
Python-[Error] percentual error = 1, analytical error = 0.08781018413800908, numerical error = 0.0878166898601482
Python-[Error] percentual error = 5, analytical error = 0.4390509206900455, numerical error = 0.43987490726751693
Python-[Error] percentual error = 10, analytical error = 0.878101841380091, numerical error = 0.8847242833350732
```

The source code involves
- migration of existing function/class
- new math (not sufficient documentation regarding the background provided in the source code, therefore unable to verify its correctness)
- some level of updates on the documentation page

Despite my limited understanding of background, I believe this PR is good enough for merging and am happy to approve :+1:

File: Framework/API/src/PeakFunctionIntegrator.cpp [hunk 1]
```diff
@@ -8,7 +8,9 @@
 
 #include "MantidAPI/FunctionDomain1D.h"
 #include "gsl/gsl_errno.h"
+#include <boost/math/distributions/students_t.hpp>
```
Comment (AndreiSavici) [inline Framework/API/src/PeakFunctionIntegrator.cpp:11]: Where is this used?

**Comment by AndreiSavici** [comment]
Is this related to any issue?

File: Framework/API/src/PeakFunctionIntegrator.cpp (hunk 1 above)
Comment (jmborr) [reply Framework/API/src/PeakFunctionIntegrator.cpp:11]: good catch. This is a leftover from a previously tried implementation of the error. I'll delete it.

Comment (jmborr) [reply Framework/API/src/PeakFunctionIntegrator.cpp]: I just deleted the unused headers.
@KedoKudo Can you please approve again after these minor change?

**Review by KedoKudo — APPROVED** [review]

**Comment by jmborr** [comment]
@KedoKudo @AndreiSavici thank you ! 🕺

**Pull request merged**